### PR TITLE
feature: Add PresenterViewHolder

### PR DESCRIPTION
**Pull request by franmontiel**
PresenterViewHolder intent is to replace WeakReference as a view wrapper on the presenter.

On Android it wrap the view in a WeakReference and also clears the view when onDestroy is called.
On iOS the implementation just wraps the view in a WeakReference.

### CODE REVIEW

**Comment by mecoFarid** [comment]
@franmontiel  Right now, it properly clears the view reference when the view is destroyed as we expect. But we probably want `Presenters` to act partly like `ViewModel`. `ViewModel` would cancel the scope, effectively stopping all jobs. But with the current implementation, we still have to do this for every presenter.

Wouldn't it be better to have `BasePresenter` that implements `CoroutineScope` and cancels that scope when the view is destroyed?

**Comment by franmontiel** [comment]
> @franmontiel Right now, it properly clears the view reference when the view is destroyed as we expect. But we probably want `Presenters` to act partly like `ViewModel`. `ViewModel` would cancel the scope, effectively stopping all jobs. But with the current implementation, we still have to do this for every presenter.
> 
> Wouldn't it be better to have `BasePresenter` that implements `CoroutineScope` and cancels that scope when the view is destroyed?

@mecoFarid take into account that we don't always need to cancel the scope. Maybe you are doing an operation that you want to finalize if you exit the screen. So from my point of view is the developer the one that should decide if they want to cancel or not.

**Comment by mecoFarid** [comment]
Good point but if we want the task to continue even after the view is destroyed probably that task shouldn't be dependent on the view's scope in the first place. Feel free to discuss it here or have a call whenever you are free.

**Comment by franmontiel** [comment]
> Good point but if we want the task to continue even after the view is destroyed probably that task shouldn't be dependent on the view's scope in the first place. Feel free to discuss it here or have a call later whenever you are free.

You might want to provide some feedback to the user if the operation is successful or not when the user is on the screen. But you could want to let the user go back and out of that context you don't care about the feedback anymore.

**Review by mecoFarid — APPROVED** [review]

**Comment by mecoFarid** [comment]
Done. Can be merged when checks pass

**Comment by joselufo** [comment]
@mecoFarid checks are not passing and the PR is on Draft. 

Is it ready to review? (as I see it on Draft)

**Comment by mecoFarid** [comment]
@joselufo Marked as ready for review. Don't know why it is failing, checking it

**Pull request merged**
